src/Instrumentation/Psr3/src/Psr3Instrumentation.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Contrib\Instrumentation\Psr3;
+
+use OpenTelemetry\API\Trace\Span;
+use OpenTelemetry\Context\Context;
+use Psr\Log\LoggerInterface;
+
+class Psr3Instrumentation
+{
+    /** @psalm-suppress ArgumentTypeCoercion */
+    public const NAME = 'psr3';
+
+    public static function register(): void
+    {
+        $pre = static function (LoggerInterface $object, ?array $params, ?string $class, string $function, ?string $filename, ?int $lineno) {
+            $context = Context::getCurrent();
+            $span = Span::fromContext($context)->getContext();
+
+            if (!$span->isValid()) {
+                return;
+            }
+
+            if ($function === 'log') {
+                $level = $params[0] ?? '';
+                $message = $params[1] ?? '';
+                $context = $params[2] ?? [];
+
+                $context['traceId'] = $span->getTraceId();
```

Comment:
This code is the same for both branches so can be merged

Reply:
Updated.